code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -88,6 +90,13 @@
 		return TRUE
 	return ..()
 
+/mob/living/simple_animal/hostile/giant_spider/expose_reagents(list/reagents, datum/reagents/source, methods=TOUCH, volume_modifier=1, show_message=TRUE)
+	. = ..()
+	for(var/datum/reagent/current_reagent as anything in reagents)
+		if(istype(current_reagent, /datum/reagent/toxin/pestkiller))
+			apply_damage(50 * volume_modifier, STAMINA, BODY_ZONE_CHEST)
```

Comment:
```suggestion
	for(var/datum/reagent/toxin/pestkiller/current_reagent in reagents)
		apply_damage(50 * volume_modifier, STAMINA, BODY_ZONE_CHEST)
```
``as anything`` just bypasses the ``istype`` check, but since we're doing it anyways, we could just make use of the loop instead.